pkg/operator/ceph/client/controller.go

```diff
@@ -375,7 +412,7 @@ func generateClientName(name string) string {
 }
 
 // updateStatus updates an object with a given status
-func (r *ReconcileCephClient) updateStatus(observedGeneration int64, name types.NamespacedName, status cephv1.ConditionType) {
+func (r *ReconcileCephClient) updateStatus(observedGeneration int64, name types.NamespacedName, status cephv1.ConditionType, cephx *cephv1.CephxStatus) {
```

Comment:
Let's wrap the internals of this in a `RetryOnConflict()` loop to help prevent possible double rotations if there is a status update conflict

Reply:
This is a corner case, but it won't be a common one. We have no better way of tracking rotation status at this time, so we have to live with the rare possiblity of double rotation.